modules/internal/ChapelBase.chpl

```diff
@@ -2303,19 +2926,43 @@ module ChapelBase {
 
 
   // non-param/non-param
-  inline operator ==(a: uint(?w), b: int(w)) {
+
+  inline operator ==(a: uint(8), b: int(8)) {
     //
     // If b's negative, these obviously aren't equal; if it's not
     // negative, it can be cast to an int
     //
```

Comment:
Bonus point: move the comment above the first ==(uint,int) and turn all four such overloads into 1-liners, like ==(int,uint) overloads are.